### ISSUE TITLE
Pushing to package managers/repositories

### ISSUE THREAD

**Issue by CrazyPython** [tqdm/tqdm.cpp#7]
Again, low priority. Just a tracker. I did a little bit of research on package managers and here's what I got:

> Un*x

- [ ] `apt-get` (`dpkg` repository)
- [ ] `pkg` (BSD)
- [ ] `rpm`/`yum`

> Windows

- [ ] **`choco`** (Chocolatey)
- [ ] OneGet 

> Mac OS X

- [ ] **`homebrew`**
- [ ] **`port`** (MacPorts)
- [ ] **`fink`**

Notes:
 - Pushing to `rpm` is equivalent to pushing to `yum` (and its flavors), `apt-rpm`, `up2date`, `urpmi`, and `ZyPP`
 - There's an (official) experimental extension to Chocolatey that makes it compatible with OneGet.

**Comment by o11c** [comment]
IMO the old-school copy-a-single-header-file-into-your-project is all we *need*. People won't be able to assume we exist as a dependency for a long time.

When $distro wants to split that out, that's their problem.

**Comment by lrq3000** [comment]
I agree, it should be very simple and portable at first, that's how `tqdm`
for python was designed at first, the packaging and all was added later on
as needed by users (so after we got some momentum).

2016-08-09 20:03 GMT+02:00 CrazyPython <notifications@github.com>:

**Comment by CrazyPython** [comment]
@lrq3000 packaging wouldn't even be necessary if the install process was simple enough.

**Comment by lrq3000** [comment]
In the python case, the packaging was useful to get released and searchable
on pypi, for unit testing and above all for submodules and CLI pipe (to be
accessible from the global env). So yeah packaging is probably not
necessary at this stage (ah I miss those days when I could include `tqdm`
as a single file in all my projects...).

2016-08-09 21:05 GMT+02:00 CrazyPython <notifications@github.com>:
